pkg/ruler/rules/recording.go

```diff
@@ -0,0 +1,203 @@
+// Copyright 2013 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rules
+
+import (
+	"context"
+	"fmt"
+	"html/template"
+	"net/url"
+	"sync"
+	"time"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/rulefmt"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/util/strutil"
+)
+
+// A RecordingRule records its vector expression into new timeseries.
+type RecordingRule struct {
+	name string
+	// Cortex-Note: this is changed to fmt.Stringer instead of a parser.Expr in order to facilitate arbitrary expressions.
+	vector fmt.Stringer
```

Comment:
Correct me if I'm wrong. The main reason we are forking this file is to change vector to be a `fmt.Stringer` type? Why not just use a `parser.StringLiteral`?

https://github.com/prometheus/prometheus/blob/72425d4e3d14d209cc3f3f6e10e3240411303399/promql/parser/ast.go#L150-L154

This would fulfill the `parser.Expr` interface and be functionally equivalent to an `fmt.Stringer` if I'm not mistaken? It would also make it so we don't have to fork the entire Prometheus `rules` package.